### PR TITLE
[AMORO-1747] Ban guava direct import when building

### LINKED ISSUE #1747: [Improvement]: Add guava check to checkstyle rules.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Add guava check to checkstyle rules. 

Do not use the com.google.guava package directly. Use iceberg guava shade instead.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klion26**
## Why are the changes needed?

Close #1747.

## Brief change log

-

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by majin1102 — APPROVED** [review]
LGTM

**Review by zhongqishang — COMMENTED** [review]
Thanks for your contribution. 
I found some that didn't use shade class because it was suppressed in `suppressions.xml`.

e.g. 
com.netease.arctic.api.config.OptimizingConfig
```
&& com.google.common.base.Objects.equal(optimizerGroup, that.optimizerGroup);
```
com.netease.arctic.trino.unkeyed.IcebergMetadata
```
import static com.google.common.base.Preconditions.checkArgument;
...
```
...

**Comment by klion26** [comment]
@zhongqishang thanks for the review. The suppressions ware introduced by #2671, and intended to be modified in #2675.
The pr here just want to replace all the remaining guava import.

I can also modify these if we need to change all guava import in this PR

**Comment by klion26** [comment]
kindly ping @zhongqishang @zhoujinsong

File: tools/maven/suppressions.xml [hunk 1]
```diff
@@ -19,183 +19,125 @@ under the License.
 -->
 
 <!DOCTYPE suppressions PUBLIC
-		"-//Puppy Crawl//DTD Suppressions 1.1//EN"
-		"http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+        "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
 
 <suppressions>
```
[the remark this answers is not in the document]
Comment (klion26) [reply tools/maven/suppressions.xml:25]: thanks for point this out, fixed

**Review by zhongqishang — APPROVED** [review]
LGTM.

**Comment by zhoujinsong** [comment]
@klion26 Thanks a lot for your contribution!

**Comment by klion26** [comment]
@zhoujinsong @zhongqishang @majin1102 thanks a lot for the review and merging!
